pkg/controller/volume/expand/expand_controller.go

```diff
@@ -123,107 +108,151 @@ func NewExpandController(
 	expc.recorder = eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "volume_expand"})
 	blkutil := volumepathhandler.NewBlockVolumePathHandler()
 
-	expc.opExecutor = operationexecutor.NewOperationExecutor(operationexecutor.NewOperationGenerator(
+	expc.operationGenerator = operationexecutor.NewOperationGenerator(
 		kubeClient,
 		&expc.volumePluginMgr,
 		expc.recorder,
 		false,
-		blkutil))
-
-	expc.resizeMap = cache.NewVolumeResizeMap(expc.kubeClient)
+		blkutil)
 
 	pvcInformer.Informer().AddEventHandler(kcache.ResourceEventHandlerFuncs{
-		UpdateFunc: expc.pvcUpdate,
-		DeleteFunc: expc.deletePVC,
+		AddFunc: expc.enqueuePVC,
+		UpdateFunc: func(old, new interface{}) {
+			oldPVC := old.(*v1.PersistentVolumeClaim)
+			oldSize := oldPVC.Spec.Resources.Requests[v1.ResourceStorage]
+
+			newPVC := new.(*v1.PersistentVolumeClaim)
+			newSize := newPVC.Spec.Resources.Requests[v1.ResourceStorage]
+			if newSize.Cmp(oldSize) > 0 {
+				expc.enqueuePVC(new)
+			}
+		},
+		DeleteFunc: expc.enqueuePVC,
 	})
 
-	expc.syncResize = NewSyncVolumeResize(syncLoopPeriod, expc.opExecutor, expc.resizeMap, kubeClient)
-	expc.pvcPopulator = NewPVCPopulator(
-		populatorLoopPeriod,
-		expc.resizeMap,
-		expc.pvcLister,
-		expc.pvLister,
-		&expc.volumePluginMgr,
-		kubeClient)
 	return expc, nil
 }
 
-func (expc *expandController) Run(stopCh <-chan struct{}) {
-	defer runtime.HandleCrash()
-	klog.Infof("Starting expand controller")
-	defer klog.Infof("Shutting down expand controller")
+func (expc *expandController) enqueuePVC(obj interface{}) {
+	pvc := obj.(*v1.PersistentVolumeClaim)
+	size := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+	statusSize := pvc.Status.Capacity[v1.ResourceStorage]
 
-	if !controller.WaitForCacheSync("expand", stopCh, expc.pvcsSynced, expc.pvSynced) {
-		return
+	if pvc.Status.Phase == v1.ClaimBound && size.Cmp(statusSize) > 0 {
```

Comment:
We need to add back the check about claimrefs

Reply:
added https://github.com/kubernetes/kubernetes/pull/75386/commits/718baa20027c7b6514d5b38f0661798491d87120#diff-6db487adca84bc74a8dc0538c8dd6463R189